### PR TITLE
Create maven.yml

### PR DESCRIPTION
**Pull request by jcschaff**
see issue #1 

set up maven build/test workflow to execute in GitHub Actions upon each commit

### CODE REVIEW

**Review by AvocadoMoon — COMMENTED** [review]
Currently it seems that whenever a push occurs to any branch, this action is initiated. This can be beneficial because no matter what all commits get tested, but it may cause redundant work to be done if for example:

    I commit some new documentation I've written. Since its a markdown file it doesn't require for the same test measures and building that a plugin modification commit may require.

There is filters that can be used with the on keyword in [GitHub's documentation](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions)

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-java-with-maven
+
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: Java CI with Maven
+
+on:
```
[the remark this answers is not in the document]
Comment (jcschaff) [reply .github/workflows/maven.yml:11]: I see that ignoring some docs is possible:
```
on:
  push:
    paths-ignore:
      - 'docs/**'
```

So far, the maven build/test actions are very fast - expensive testing or important side effects will be limited to future actions triggered on PR merge and on Release.

[the remark this answers is not in the document]
Comment (jcschaff) [reply .github/workflows/maven.yml:11]: but I'll simplify the pattern to '*'

[the remark this answers is not in the document]
Comment (jcschaff) [reply .github/workflows/maven.yml:11]: @AvocadoMoon thank you for your comments.

[the remark this answers is not in the document]
Comment (AvocadoMoon) [reply .github/workflows/maven.yml:11]: @jcschaff Thank you for the response.

**Review by AvocadoMoon — APPROVED** [review]
Everything seems fine

**Pull request merged**